datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -231,8 +260,14 @@ protected void restartDeadTasks() {
         if (isTaskDead(connectorTaskEntry)) {
           _logger.warn("Detected that the kafka connector task is not running for datastream task {}. Restarting it",
               datastreamTask.getDatastreamTaskName());
-          boolean stopped = stopTask(datastreamTask, connectorTaskEntry);
-          if (stopped) {
+          // Connector task attempt stop will be set to false if stopTask is successful in stopping the connector task.
+          // If stopping the task fails the flag will remain True to attempt another stop before creating starting
+          // another connector task.
```

Comment:
I'm not sure why the comments talk about setting things false or true here. It looks like stopTask() returns the DatastreamTask, so perhaps modify the comments to talk about null/not-null instead? Or maybe I didn't understand the comments too well, so can you explain what you're trying to say here?